runtime/stringbuf.c

```diff
@@ -534,6 +534,15 @@ rsRetVal cstrTrimTrailingWhiteSpace(cstr_t *pThis)
 	if(i != (int) pThis->iStrLen) {
 		pThis->iStrLen = i;
 		pThis->pBuf[pThis->iStrLen] = '\0'; /* we always have this space */
+
+		if(pThis->pszBuf != NULL) {
+			/* in this case, we adjust the psz representation
+			 * by writing a new \0 terminator - this is by far
+			 * the fastest way and outweights the additional memory
+			 * required.
+			 */
+			 pThis->pszBuf[pThis->iStrLen] = '\0';
```

Comment:
Regarding this code and the twin place in `rsCStrTruncate`: If you don't reset `pszBuf` when you append data to `pBuf` (as I have suggested), you can actually write the `'\0'` byte into unknown memory after `pszBuf`:
```C
cstr_t *thisLine = NULL;
rsCStrConstructFromszStr(&thisLine, "foo"); /* pBuf = "foo" */
rsCStrGetSzStrNoNULL(thisLine);                /* pszBuf = "foo\0" */
rsCStrAppendStr(thisLine, "bar ");                 /* pBuf  = "foobar ", pszBuf = "foo\0" */
cstrTrimTrailingWhiteSpace(thisLine);           /* pBuf = "foobar", plus writing \0 after the end of pszBuf */
```
The last call will actually do `thisLine->pszBuf[6] = '\0'`, but `pszBuf` is only 4 bytes long ("foo\0").

(In my alternative implementation I check that `pszBuf`is not stale before writing the null byte.)